metacat-common/src/main/java/com/netflix/metacat/common/QualifiedName.java

```diff
@@ -550,33 +593,88 @@ public String toString() {
         return qualifiedName;
     }
 
+    /**
+     *
+     * Checks if a CharSequence is empty ("") or null.
+     */
+    private static boolean isNullOrEmpty(@Nullable final CharSequence cs) {
+        return cs == null || cs.length() == 0;
+    }
+
     /**
      * Type of the connector resource.
      */
     public enum Type {
         /**
          * Catalog type.
          */
-        CATALOG,
+        CATALOG("CATALOG"),
 
         /**
          * Database type.
          */
-        DATABASE,
+        DATABASE("DATABASE"),
 
         /**
          * Table type.
          */
-        TABLE,
+        TABLE("TABLE"),
 
         /**
          * Partition type.
          */
-        PARTITION,
+        PARTITION("PARTITION"),
```

Comment:
By default, Enum.name() will return the string value as it is defined in the enum. So we do not have to explicitly set it.

Reply:
This calls constructor with value "PARTITION"., which is used in the converter  in tagController. 
        public void setAsText(final String text) throws IllegalArgumentException {
            super.setValue(QualifiedName.Type.fromValue(text));
        }

Reply:
You could use valueOf.